lib/connection.cpp

```diff
@@ -2395,8 +2395,8 @@ bool Connection::Private::createOlmSession(const QString& targetUserId,
                         << recipientCurveKey << session.error();
         return false;
     }
-    saveSession(*session, recipientCurveKey);
-    olmSessions[recipientCurveKey].push_back(std::move(*session));
+    saveSession(*session, QString::fromUtf8(recipientCurveKey));
+    olmSessions[QString::fromUtf8(recipientCurveKey)].push_back(std::move(*session));
```

Comment:
You can clearly see just a few lines above that `recipientCurveKey` is in Latin-1.
```suggestion
    saveSession(*session, QString::fromLatin1(recipientCurveKey));
    olmSessions[QString::fromLatin1(recipientCurveKey)].push_back(std::move(*session));
```